giantswarm/cloud-provider-cloud-director-app
# Initial dev branch

This PR:

- adds the helm charts of the CSI (1.2.0) and CPI (1.1.1) for VMware Cloud Director (VCD)

### Testing

Description on how cloud-provider-cloud-director-app can be tested.

- Deploy a VCD cluster to the test provider (Ikoula, details in lastpass) or use an existing test cluster. The current [cluster-cloud-director](https://github.com/giantswarm/cluster-cloud-director) automatically installs CPI and CSI within cloud-init so you'll need to remove them.
- Uninstall the CSI & CPI if already installed `kubectl delete -f <folder>` where folder is https://github.com/vmware/cloud-director-named-disk-csi-driver/tree/main/manifests and https://github.com/vmware/cloud-provider-for-cloud-director/tree/1.1.x/manifests.
- Edit the top chart's values.yaml and install it.
- You can try the CSI (follow readme) but not the CPI for now as it needs an extra public IP (WIP upstream). Just check the logs of `vmware-cloud-director-ccm-xxxxxxxxx-xxxxxx`

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.
